File: CSharp/6kyu/FindTheOddInt.cs
using System.Linq;
using System;

namespace CSharp._6Kyu
{
    public static partial class Kata6
    {
        public static int find_it_fifi(int[] seq)
        {
            return seq.FirstOrDefault(s => seq.Count(a => a == s) % 2 != 0);
        }

        public static int find_it(int[] seq)
        {
            return seq.GroupBy(x => x).Single(g => g.Count() % 2 == 1).Key;
        }
    }
}

File: Python/6kyu/FindTheOddInt.py
def find_it_fifi(seq):
    return [s for s in seq if seq.count(s) % 2 != 0][0]


def find_if(seq):
    for s in seq:
        if seq.count(s) % 2 != 0:
            return s


result = find_it_fifi(
    [20, 1, -1, 2, -2, 3, 3, 5, 5, 1, 2, 4, 20, 4, -1, -2, 5])
print(result)
